velox/functions/lib/aggregates/noisy_aggregation/sketch/SfmSketch.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include "folly/hash/MurmurHash.h"
+#include "velox/common/base/BitUtil.h"
+#include "velox/common/memory/HashStringAllocator.h"
+#include "velox/functions/lib/aggregates/noisy_aggregation/sketch/SecureRandomizationStrategy.h"
+
+namespace facebook::velox::functions::aggregate {
+
+/// For technical details of SfmSketch, please refer to the paper
+/// <a href="https://arxiv.org/pdf/2302.02056.pdf">Sketch-Flip-Merge: Mergeable
+/// Sketches for Private Distinct Counting</a>.
+class SfmSketch {
+ public:
+  // Constructor to create a non-private sketch. This constructor does not set
+  // the size of the sketch because we don't know the size of the sketch until
+  // we decoded the input data, so we will set the size in the initialize
+  // function.
+  explicit SfmSketch(HashStringAllocator* allocator);
+
+  // Set the sketch size. The size is determined by the number of buckets and
+  // the precision. This function should be called before any other functions
+  // except the constructor.
+  void initialize(uint32_t buckets, uint32_t precision);
```

Comment:
Do we require buckets to be power of two? Let's document that? What are requirements for 'precision'?